back-end/src/sciler/handler/handler.go

```diff
@@ -166,68 +171,181 @@ func (handler *Handler) onInstructionMsg(raw Message) {
+// test device is instructed when the test button of a device is clicked in the front-end
+func (handler *Handler) onTestDevice(deviceID string, instructor string) {
+	handler.sendInstruction(deviceID, []map[string]string{{
+		"instruction":   "test",
+		"instructed_by": instructor,
+	}})
+}
+
+// onResetAll is the function to process the instruction `finish rule`
+// finish rule is instructed then the "voer uit" button of a rule is clicked in the front-end
+func (handler *Handler) onFinishRule(ruleID string) {
+	rule, ok := handler.Config.RuleMap[ruleID]
+	if !ok {
+		logger.Errorf("could not find rule with id %s in map", ruleID)
+	} else {
+		rule.Execute(handler)
+	}
+	handler.sendEventStatus()
+}
+
+// onHint is the function to process the instruction `hint`
+// hint in instructed when the front-end submits a hint
+func (handler *Handler) onHint(hint string, instructor string) {
+	handler.sendInstruction("hint", []map[string]string{{
+		"instruction":   "hint",
+		"value":         hint,
+		"instructed_by": instructor,
+	}})
+}
+
+// onCheckConfig is the function to process the instruction `check config`
+// check the config and sends a message containing all errors it could find
+func (handler *Handler) onCheckConfig(configToRead interface{}) {
+	message := Message{
+		DeviceID: "back-end",
+		TimeSent: time.Now().Format("02-01-2006 15:04:05"),
+		Type:     "config",
+		Contents: map[string][]string{"errors": handler.checkConfig(configToRead)},
+	}
+	jsonMessage, _ := json.Marshal(&message)
+	handler.Communicator.Publish("front-end", string(jsonMessage), 3)
+}
+
+// processConfig reads the config in.
+// If action is "use" then the message must tell the config a new config is now used and put it to use
+func (handler *Handler) onUseConfig(configToRead interface{}, fileName string) {
+	// check if the config can be used, if so use and send message
+	if len(handler.checkConfig(configToRead)) == 0 {
+		handler.useConfig(configToRead, fileName)
+		message := Message{
+			DeviceID: "back-end",
+			TimeSent: time.Now().Format("02-01-2006 15:04:05"),
+			Type:     "new config",
+			Contents: map[string]string{"name": fileName},
+		}
+		jsonMessage, _ := json.Marshal(&message)
+		handler.Communicator.Publish("front-end", string(jsonMessage), 3)
+	}
+}
+
+// useConfig uses new config and save this config to file
+// warning this config should be checked first before using the config
+func (handler *Handler) useConfig(configToRead interface{}, fileName string) {
+	jsonBytes, _ := json.Marshal(configToRead)
+	newConfig, _ := config.ReadJSON(jsonBytes)
+	dir, _ := os.Getwd()
+	fullFileName := filepath.Join(dir, "back-end", "resources", "production", fileName)
+	err := ioutil.WriteFile(fullFileName, jsonBytes, 0644)
+	if err != nil {
+		logger.Error(err)
+	}
+	handler.Config = newConfig
+	handler.ConfigFile = fullFileName
+	handler.SendSetup()
+}
+
+// connected is a method that sets a device to connected
```

Comment:
maybe put this in a different spot to keep code structured

Reply:
Moved this method to handler_helper